--- src/Makefile.am.orig	2012-02-07 22:39:22.384339300 -0500
+++ src/Makefile.am	2012-02-07 23:23:07.990327813 -0500
@@ -5,33 +5,34 @@
 
 canondr_com_usb_PROGRAMS=canondr_com_usb
 canondr_com_usb_SOURCES=canondr_com_usb.c
-canondr_com_usb_LDADD=$(SANEIDIR)/libsanei.a
-canondr_com_usbdir=$(prefix)/lib/canondr
+canondr_com_usb_LDADD=libsane-canondr.la
+
+canondr_com_usbdir=$(libdir)/canondr
 canondr_com_usbPROGRAMS_INSTALL=$(install_sh_PROGRAM) -m 4755
 
-SANESRC=../../sane-backends-1.0.19
+SANESRC=../sane
 SANEIDIR=$(SANESRC)/sanei
-lib_LTLIBRARIES = libsane-canondr.la
+sanelib_LTLIBRARIES = libsane-canondr.la
+sanelibdir=$(libdir)/sane
+
 libsane_canondr_la_SOURCES= \
 	Canon_DR.h\
-	client.c
-libdir= $(prefix)/lib/sane
+	client.c \
+	$(SANEIDIR)/sanei_net.c\
+	$(SANEIDIR)/sanei_wire.c\
+	$(SANEIDIR)/sanei_config.c\
+	$(SANEIDIR)/sanei_codec_bin.c\
+	$(SANEIDIR)/sanei_constrain_value.c\
+	$(SANEIDIR)/sanei_init_debug.c\
+	$(SANEIDIR)/sanei_usb.c
 
 INCLUDES= -I$(SANESRC)/include
 DEFS=\
 	-DV_MAJOR=$(V_MAJOR)\
 	-DV_MINOR=$(V_MINOR)\
 	-DBUILD=$(BUILD)\
-	-DDRBINPATH=\"$(prefix)/lib/canondr\"\
+	-DDRBINPATH=\"$(libdir)/canondr\"\
 	-DSANECONFDIR=\"$(sysconfdir)/sane.d\"
 
 libsane_canondr_la_LDFLAGS= -shared \
 	-version-info 1:0:0
-libsane_canondr_la_LIBADD=\
-	$(SANEIDIR)/sanei_net.lo\
-	$(SANEIDIR)/sanei_wire.lo\
-	$(SANEIDIR)/sanei_config.lo\
-	$(SANEIDIR)/sanei_codec_bin.lo\
-	$(SANEIDIR)/sanei_constrain_value.lo\
-	$(SANEIDIR)/sanei_init_debug.lo\
-	$(SANEIDIR)/sanei_usb.lo
