diff -ur mocker-1.0.orig/setup.py mocker-1.0/setup.py
--- mocker-1.0.orig/setup.py	2010-06-20 19:14:18.000000000 -0400
+++ mocker-1.0/setup.py	2010-08-10 12:12:34.981896691 -0400
@@ -7,6 +7,11 @@
 except ImportError:
     from distutils.core import setup, Extension
 
+try:
+    from distutils.command.build_py import build_py_2to3 as build_py
+except ImportError:
+    from distutils.command.build_py import build_py
+
 
 if os.path.isfile("MANIFEST"):
     os.unlink("MANIFEST")
@@ -33,4 +38,5 @@
         "Topic :: Software Development :: Testing",
         "Topic :: Software Development :: Libraries :: Python Modules",
     ],
+    cmdclass = {'build_py': build_py},
 )
