--- setup.py.orig	2009-01-21 17:58:53.503249992 -0500
+++ setup.py	2009-01-21 17:59:04.577667440 -0500
@@ -66,7 +66,6 @@
             "dependency_links.txt = setuptools.command.egg_info:overwrite_arg",
         ],
         "console_scripts": [
-             "easy_install = setuptools.command.easy_install:main",
              "easy_install-%s = setuptools.command.easy_install:main"
                 % sys.version[:3]
         ],
