From d4afd6495398155db82f8dc94c18b9ab13720c70 Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@edgedb.com>
Date: Mon, 27 Jun 2022 14:32:09 -0700
Subject: [PATCH] stb include fix

---
 meson.build | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 1311784..32568fe 100644
--- a/meson.build
+++ b/meson.build
@@ -45,8 +45,6 @@ sdl_dep = dependency('SDL2')
 pipewire_dep = dependency('libpipewire-0.3', required: get_option('pipewire'))
 librt_dep = cppc.find_library('rt', required : get_option('pipewire'))
 
-stb_dep = dependency('stb')
-
 wlroots_dep = dependency(
   'wlroots',
   version: ['>= 0.15.0', '< 0.16.0'],
@@ -142,7 +140,8 @@ executable(
     dep_x11, dep_xdamage, dep_xcomposite, dep_xrender, dep_xext, dep_xfixes,
     dep_xxf86vm, dep_xres, drm_dep, wayland_server, wayland_protos,
     xkbcommon, thread_dep, sdl_dep, wlroots_dep,
-    vulkan_dep, liftoff_dep, dep_xtst, cap_dep, pipewire_dep, librt_dep, stb_dep,
+    vulkan_dep, liftoff_dep, dep_xtst, cap_dep, pipewire_dep, librt_dep,
   ],
   install: true,
+  include_directories: include_directories('/usr/include/stb'),
 )
-- 
2.35.1

