diff --git a/bios/rombios32.c b/bios/rombios32.c
index 4dea066..acc5661 100755
--- a/bios/rombios32.c
+++ b/bios/rombios32.c
@@ -1776,6 +1776,13 @@ struct smbios_type_127 {
 	struct smbios_structure_header header;
 } __attribute__((__packed__));
 
+
+/* SMBIOS type 218 -- OEM-Specific */
+struct smbios_type_218 {
+	struct smbios_structure_header header;
+} __attribute__((__packed__));
+
+
 static void
 smbios_entry_point_init(void *start,
                         uint16_t max_structure_size,
@@ -2064,6 +2071,38 @@ smbios_type_32_init(void *start)
     return start+2;
 }
 
+/* Type 218 -- OEM ID */
+static void *
+smbios_type_218_init(void *start)
+{
+    struct smbios_type_218 *p = (struct smbios_type_218 *)start;
+
+    p->header.type = 218;
+    p->header.length = sizeof(struct smbios_type_218);
+    p->header.handle = 0xDA00;
+
+    start += sizeof(struct smbios_type_218);
+
+    *((uint16_t *)start) = 0;
+
+#ifndef OEMID_STR
+    return start + 2;
+#else
+    if ((unsigned int)start > OEMID_ADDR)
+        return start + 2;
+
+    p->header.length += OEMID_ADDR + sizeof(OEMID_STR) - (unsigned int)start;
+
+    memset(start, 0, OEMID_ADDR - (unsigned int)start);
+    memcpy((char *)OEMID_ADDR, OEMID_STR, sizeof(OEMID_STR));
+
+    start = (char *)OEMID_ADDR + sizeof(OEMID_STR);
+    *((uint16_t *)start) = 0;
+
+    return start+2;
+#endif
+}
+
 /* Type 127 -- End of Table */
 static void *
 smbios_type_127_init(void *start)
@@ -2125,6 +2164,7 @@ void smbios_init(void)
     }
 
     add_struct(smbios_type_32_init(p));
+    add_struct(smbios_type_218_init(p));
     add_struct(smbios_type_127_init(p));
 
 #undef add_struct
