diff -ur flup-1.0.1.orig/setup.py flup-1.0.1/setup.py
--- flup-1.0.1.orig/setup.py	2008-07-22 14:05:11.000000000 -0400
+++ flup-1.0.1/setup.py	2009-01-21 18:03:27.246681326 -0500
@@ -1,5 +1,10 @@
 # Bootstrap setuptools
 from ez_setup import use_setuptools
+try:
+    from distutils.command.build_py import build_py_2to3 as build_py
+except ImportError:
+    from distutils.command.build_py import build_py
+
 use_setuptools()
 
 from setuptools import setup, find_packages
@@ -37,4 +42,5 @@
     'Topic :: Internet :: WWW/HTTP :: WSGI :: Server',
     'Topic :: Software Development :: Libraries :: Python Modules',
     ],
+    cmdclass={'build_py': build_py},
     )
