From e0c56f1ddb63f2c646b0dae32da47a06678f41c1 Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@edgedb.com>
Date: Sun, 3 Jul 2022 12:52:45 -0700
Subject: [PATCH] Build fixes

---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 5a363fd..7a9d62c 100644
--- a/meson.build
+++ b/meson.build
@@ -276,8 +276,7 @@ endif
 
 if get_option('mangoapp') or get_option('mangoapp_layer')
   glfw3_dep = dependency('glfw3')
-  json_sp = subproject('nlohmann_json')
-  json_dep = json_sp.get_variable('nlohmann_json_dep')
+  json_dep = dependency('nlohmann_json')
 endif
 
 subdir('src')
-- 
2.35.1

