Index: setup.py
===================================================================
--- setup.py	(revision 54)
+++ setup.py	(working copy)
@@ -3,6 +3,11 @@
 except ImportError:
     from distutils.core import setup
 
+try:
+    from distutils.command.build_py import build_py_2to3 as build_py
+except ImportError:
+    from distutils.command.build_py import build_py
+
 setup(name = "ply",
             description="Python Lex & Yacc",
             long_description = """
@@ -27,4 +32,5 @@
             maintainer_email = "dave@dabeaz.com",
             url = "http://www.dabeaz.com/ply/",
             packages = ['ply'],
+	    cmdclass = {'build_py': build_py}
             )
