diff --git a/hald/linux/addons/addon-dell-backlight.cpp b/hald/linux/addons/addon-dell-backlight.cpp
index fa27fac..dd9b1e6 100644
--- a/hald/linux/addons/addon-dell-backlight.cpp
+++ b/hald/linux/addons/addon-dell-backlight.cpp
@@ -111,7 +111,7 @@ write_backlight (u32 newBacklightValue, dbus_bool_t onAC)
 	string password(""); /* FIXME: Implement password support */
 
 #ifdef HAVE_SMBIOS_2_0_3
-	if (hasPass) {
+	if (0) {
 		HAL_WARNING(("Setting brightness via SMI is not supported with a BIOS password"));
 		return;
 	}
