Only in feedparser-5.0: build
diff -ur feedparser-5.0.orig/feedparser/feedparser.py feedparser-5.0/feedparser/feedparser.py
--- feedparser-5.0.orig/feedparser/feedparser.py	2011-01-23 04:08:37.000000000 -0500
+++ feedparser-5.0/feedparser/feedparser.py	2011-01-26 11:32:20.953275003 -0500
@@ -135,7 +135,13 @@
 #ACCEPTABLE_URI_SCHEMES = ()
 
 # ---------- required modules (should come with any Python distribution) ----------
-import sgmllib, re, sys, copy, urlparse, time, types, cgi, urllib, urllib2, datetime
+import re, sys, copy, urlparse, time, types, cgi, urllib, urllib2, datetime
+
+try:
+    import smgllib
+except ImportError:
+    import sgmllib3 as sgmllib
+
 try:
     from io import BytesIO as _StringIO
 except ImportError:
diff -ur feedparser-5.0.orig/setup.py feedparser-5.0/setup.py
--- feedparser-5.0.orig/setup.py	2011-01-23 03:06:53.000000000 -0500
+++ feedparser-5.0/setup.py	2011-01-26 11:40:13.664275002 -0500
@@ -1,23 +1,36 @@
-from distutils.core import setup
-import imp
-import os.path
+import os
+import sys
+
+from distutils.core import setup, Command
+try:
+    from distutils.command.build_py import build_py_2to3 as build_py
+except ImportError:
+    from distutils.command.build_py import build_py
+
+
+class Build(build_py):
+    def run_2to3(self, files):
+        files = files[:]
+        try:
+            files.remove('build/lib/sgmllib3.py')
+        except ValueError:
+            pass
+        result = super(Build, self).run_2to3(files)
+        return result
 
-path = os.path.join('feedparser', 'feedparser.py')
-feedparser = imp.load_source('feedparser', path)
 
 setup(
     name = 'feedparser',
-    version = feedparser.__version__,
+    version = '5.0',
     description = 'Universal feed parser, handles RSS 0.9x, RSS 1.0, '
                   'RSS 2.0, CDF, Atom 0.3, and Atom 1.0 feeds',
-    long_description = feedparser.__doc__.strip(),
     author = 'Mark Pilgrim',
     author_email = 'mark@diveintomark.org',
     url = 'http://feedparser.org/',
     download_url = 'http://code.google.com/p/feedparser/',
     platforms = ['POSIX', 'Windows'],
     package_dir = {'': 'feedparser'},
-    py_modules = ['feedparser'],
+    py_modules = ['feedparser'] + (['sgmllib3'] if sys.version_info >= (3,) else []),
     keywords = ['atom', 'cdf', 'feed', 'parser', 'rdf', 'rss'],
     classifiers = [
         'Development Status :: 5 - Production/Stable',
@@ -35,5 +48,6 @@
         'Programming Language :: Python :: 3.1',
         'Topic :: Software Development :: Libraries :: Python Modules',
         'Topic :: Text Processing :: Markup :: XML',
-    ]
+    ],
+    cmdclass={'build_py': Build}
 )
